src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
We support only es6 for now, see there: https://github.com/dial-once/node-bunnymq#installation
Because we use `let`, etc.

And this is not an es6 class, it is es5 compliant homemade class (bundled in the PR). If you have another idea to fullfil a promise outside of its own scope, would be glad to replace it!

We had to remove its usage because it has been removed from node7, see #52